### PR TITLE
Improve retention and expiry documentation (WIP)

### PR DESCRIPTION
**Pull request by lucperkins**

### CODE REVIEW

File: site/docs/latest/getting-started/ConceptsAndArchitecture.md [hunk 1]
```diff
@@ -266,13 +266,35 @@ In BookKeeper, *journal* files contain BookKeeper transaction logs. Before makin
...
+## Message retention and expiry
+
+By default, Pulsar message {% popover brokers %}:
+
+* immediately delete *all* messages that have been {% popover acknowledged %} by a {% popover consumer %}, and
+* [persistently store](#persistent-storage) all unacknowledged messages in a message backlog.
+
+Pulsar has two features, however, that enable you to override this default behavior:
+
+* Message **retention** enables you to store messages that have been acknowledged by a consumer
```
Comment (merlimat) [inline site/docs/latest/getting-started/ConceptsAndArchitecture.md]: I think one source of confusion here was between using the term consumer. Some people were confused about that if I have multiple subscriptions, all the subscriptions will have to "acknowledge" the messages before the data can be considered for deletion. 

Also, we should add that retention also works when there are no subscriptions to retain the data.

File: site/docs/latest/getting-started/ConceptsAndArchitecture.md [hunk 2]
```diff
@@ -266,13 +266,35 @@ In BookKeeper, *journal* files contain BookKeeper transaction logs. Before makin
...
 
 ## Multi-tenancy
 
-Pulsar was created from the ground up as a {% popover multi-tenant %} system. To support multi-tenancy, Pulsar has a concept of {% popover properties %}. Properties can be spread across {% popover clusters %} and can each have their own [authentication and authorization](../../admin/Authz) scheme applied to them. They are also the administrative unit at which [storage quotas](TODO), [message TTL](TODO), and [isolation policies](TODO) can be managed.
+Pulsar was created from the ground up as a {% popover multi-tenant %} system. To support multi-tenancy, Pulsar has a concept of {% popover properties %}. Properties can be spread across {% popover clusters %} and can each have their own [authentication and authorization](../../admin/Authz) scheme applied to them. They are also the administrative unit at which [storage quotas](TODO), [message TTL](../../advanced/RetentionExpiry#time-to-live-ttl), and [isolation policies](TODO) can be managed.
 
 The multi-tenant nature of Pulsar is reflected mostly visibly in topic URLs, which have this structure:
```
Comment (merlimat) [inline site/docs/latest/getting-started/ConceptsAndArchitecture.md]: One comment on the diagram below, for message expiry, can we use a different color for the section "Within the applied TTL"? The reason is that, technically, the TTL is just auto-acknowledging messages that were not acked withing the configured time.

File: site/docs/latest/getting-started/ConceptsAndArchitecture.md (hunk 1 above)
Comment (lucperkins) [reply site/docs/latest/getting-started/ConceptsAndArchitecture.md]: @merlimat Is it always the case that all consumers need to acknowledge a message? Or does that vary based on the subscription type?

Comment (merlimat) [reply site/docs/latest/getting-started/ConceptsAndArchitecture.md]: Well, if you have a subscription `s1` with 1 or more consumer that acks everything, this won't necessary mean that the data is going to be deleted. There can be a subscription `s2` where the consumer is disconnected or lagging, and that would keep the data around, until `s2` catches up.

**Pull request merged**